File: src/tmp/Python/array/spiral_matrix_ii.py
from typing import *

def generateMatrix(n: int) -> List[List[int]]:
    matrix = [[0 for _ in range(n)] for _ in range(n)]
    direct = [[0, 1], [1, 0], [0, -1], [-1, 0]]
    row, col, dir = 0, 0, 0
    for i in range(n * n):
        matrix[row][col] = i + 1
        next_row, next_col = row + direct[dir][0], col + direct[dir][1]
        if next_row < 0 or next_row >= n or next_col < 0 or next_col >= n or matrix[next_row][next_col] != 0:
            dir = (dir + 1) % 4
        row, col = row + direct[dir][0], col + direct[dir][1]
    
    return matrix

File: core/tmp/Java/src/com/dbc/algorithm/Array/SpiralMatrixII.java
package com.dbc.algorithm.Array;

public class SpiralMatrixII {
    public int[][] generateMatrix(int n) {
        int[][] matrix = new int[n][n];
        int[][] direct = new int[][]{{0, 1}, {1, 0}, {0, -1}, {-1, 0}};
        int row = 0, col = 0, dir = 0;
        for (int i = 0; i < n * n; i++) {
            matrix[row][col] = i + 1;
            int nextRow = row + direct[dir][0];
            int nextCol = col + direct[dir][1];
            if (nextRow < 0 || nextRow >= n || nextCol < 0 || nextCol >= n || matrix[nextRow][nextCol] != 0) {
                dir = (dir + 1) % 4;
            }
            row += direct[dir][0];
            col += direct[dir][1];
        }
        return matrix;
    }
}
